src/types/tcpproxy.go

```diff
@@ -0,0 +1,43 @@
+package types
+
+import "net"
+
+func connToChan(mtu uint64, conn net.Conn) chan []byte {
+	c := make(chan []byte)
+	go func() {
+		for {
+			b := make([]byte, mtu)
```

Comment:
would this chug gc less if we used a global `sync.Pool` here?

Reply:
actually, what if we used `io.Pipe()` and `io.CopyN()` in `ProxyTCP()`?

Reply:
actually, you don't even need `io.Pipe()` you could just do `io.CopyBuffer` and flip which connection is passed in. no need to use a channel. unless i am missing something.